GeoLib/Plane.h

```diff
@@ -0,0 +1,118 @@
+/**
+ * \brief  Definition of the class Plane - a plane in 3d space described in
+ * Hessian normal form.
+ *
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef OGS_PLANE_H
+#define OGS_PLANE_H
+
+#include <utility>
+
+#include "MathLib/Vector3.h"
+
+namespace GeoLib
+{
+
+class Plane final
+{
+public:
+	Plane(MathLib::Vector3 const& unit_normal, double d)
+		: _unit_normal(unit_normal), _oriented_distance(d)
```

Comment:
Maybe it's better to check if unit_normal really has unit length, and if not, to scale.
`getDistance()` relies on that feature.